Add the Generated Connector Code and Documentation (#2)
ballerina-platform/module-ballerinax-hubspot.automation.actions: docs/spec/sanitations.md

The OpenAPI specification is obtained from (TODO: Add source link).
These changes are done in order to improve the overall usability, and as workarounds for some known language limitations.

1. **Change the `url` property of the `servers` object**:
   - **Original**: `https://api.hubapi.com`
   - **Updated**: `https://api.hubapi.com/automation/v4/actions`

>    - **Reason**: This change is made to ensure that all API paths are relative to the versioned base URL (`/4`), which improves the consistency and usability of the APIs.

Reviewer: ```suggestion
   - **Reason**: This change is made to ensure that all API paths are relative to the URL (`/automation/v4/actions`), which improves the consistency and usability of the APIs.
```

Author: done -https://github.com/ballerina-platform/module-ballerinax-hubspot.automation.actions/pull/2/commits/a6693307e6229fb15e40b7b4158f301be9a0dda6

2. **Update API Paths**:

>    - **Original**: Paths included the version prefix in each endpoint (e.g., `/2/compliance/jobs`).

Reviewer: ```suggestion
   - **Original**: Paths shared a common segment across all resource endpoints
```

Author: done -https://github.com/ballerina-platform/module-ballerinax-hubspot.automation.actions/pull/2/commits/a6693307e6229fb15e40b7b4158f301be9a0dda6

>    - **Updated**: Paths are modified to remove the version prefix from the endpoints, as it is now included in the base URL. For example:

Reviewer: ```suggestion
   - **Updated**: Common paths segment is removed from the resource endpoints, as it is now included in the base URL. For example:
```

Author: done -https://github.com/ballerina-platform/module-ballerinax-hubspot.automation.actions/pull/2/commits/a6693307e6229fb15e40b7b4158f301be9a0dda6

     - **Original**: `/automation/v4/actions`
     - **Updated**: `/`
   - **Reason**: This modification simplifies the API paths, making them shorter and more readable. It also centralizes the versioning to the base URL, which is a common best practice.

## OpenAPI cli command

The following command was used to generate the Ballerina client from the OpenAPI specification. The command should be executed from the repository root directory.

```bash

> bal openapi -i docs/spec/openapi.json --mode client -o ballerina

Reviewer: Let's add the license part also
